### PR TITLE
Ca 27 put /kingdom/troop/{lvl}

### PR DESCRIPTION
**Pull request by Kalloaa**
thank you happy birthday

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java [hunk 1]
```diff
@@ -206,5 +206,16 @@ public Troop upgradeTroop(Troop troop)
     this.updateKingdom(kingdom);
     return troop;
   }
+
+  @Override
+  public Troop upgradeRandomTroop(String username, int lvlToUpgrade)
```
Comment (csadesz) [inline src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java]: There is an upgradeRandomTroop, and an upgradeTroop method as well. There should be only one upgradeTroop method. Try to merge these.

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java [hunk 2]
```diff
@@ -180,8 +181,14 @@ public Troop createTroop(String troopType, String username)
...
-      throws MissingBuildingException, InsufficientGoldException {
+  public Troop upgradeTroop(String username, int lvlToUpgrade)
+      throws InsufficientGoldException, MissingBuildingException {
+    List<Troop> troops = kingdomRepository.findByAppUserUsername(username).getTroops();
+    Troop troop = troops.stream()
+        .filter(x -> lvlToUpgrade == x.getLevel())
+        .findFirst()
+        .orElseThrow(() -> new MissingTroopException("No troop of that level"));
+
     Kingdom kingdom = this.updateKingdom(troop.getKingdom());
```
Comment (csadesz) [inline src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java]: In this case we are going to the database (for Kingdom) twice

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java [hunk 3]
```diff
@@ -180,8 +181,14 @@ public Troop createTroop(String troopType, String username)
   }
 
   @Override
-  public Troop upgradeTroop(Troop troop)
-      throws MissingBuildingException, InsufficientGoldException {
+  public Troop upgradeTroop(String username, int lvlToUpgrade)
+      throws InsufficientGoldException, MissingBuildingException {
+    List<Troop> troops = kingdomRepository.findByAppUserUsername(username).getTroops();
+    Troop troop = troops.stream()
+        .filter(x -> lvlToUpgrade == x.getLevel())
```
Comment (ZaitzeV16) [inline src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java]: Shouldn't be x named e.g. troop?

File: src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java (hunk 2 above)
Comment (Kalloaa) [reply src/main/java/com/greenfoxacademy/chamaaragonite/kingdom/KingdomServiceImpl.java]: one of them is to find the kingdom, the other is to update its resources

**Pull request merged**